back/engines/commercial/insights/spec/acceptance/tasks/category_suggestions_spec.rb

```diff
@@ -25,8 +24,32 @@
 
   before { header 'Content-Type', 'application/json' }
 
-  let(:view) { create(:view) }
-  let(:view_id) { view.id }
+  # In summary, we set up a view with 2 categories and 3 tasks
+  # (2 tasks with category 1 and 1 task with category 2), and 
+  # another task not related to that view
+  let_it_be(:view) { create(:view) }
+  let_it_be(:view_id) { view.id }
+  let_it_be(:c1) { create(:category, view: view) }
+  let_it_be(:c2) { create(:category, view: view) }
+  let_it_be(:tasks_c1) { create_list(:zsc_task, 2, categories: [c1]) }
+  let_it_be(:tasks_c2) { create_list(:zsc_task, 1, categories: [c2]) }
+  let_it_be(:another_task) { create(:zsc_task) } # task in another view
+
+  def self.parameter_categories
+    parameter(
+      :categories, 'An array of category identifiers',
+      required: false,
+      type: :array, items: { type: :string }
+    )
+  end
+
+  def self.parameters_input_filters
+    with_options scope: :inputs, required: false do
+      parameter :categories, 'Filter inputs by categories (union)', type: :array, items: { type: :string }
+      parameter :keywords, 'Filter inputs by keywords (identifiers of keyword nodes)', type: :array, items: { type: :string }
+      parameter :processed, 'Filter inputs by processed status', type: :boolean
+    end
+  end
```

Comment:
Three endpoints have those parameters in common, so I DRYed it up by refactoring them into a method.